src/status_im/ui/screens/home/views/inner_item.cljs

```diff
@@ -103,20 +103,20 @@
 
 (defn home-list-item [[_ home-item]]
   (let [{:keys [chat-id chat-name color online group-chat
-                public? contact timestamp last-message]}
+                public? timestamp last-message]}
         home-item
         private-group?      (and group-chat (not public?))
-        public-group?       (and group-chat public?)
-        ;;TODO (perf) move to event
-        truncated-chat-name (utils/truncate-str chat-name 30)]
+        public-group?       (and group-chat public?)]
     [list-item/list-item
      {:icon                      [chat-icon.screen/chat-icon-view-chat-list
-                                  contact group-chat truncated-chat-name color online false]
+                                  chat-id group-chat chat-name color online false]
       :title-prefix              (cond
                                    private-group? :main-icons/tiny-group
                                    public-group? :main-icons/tiny-public
                                    :else nil)
-      :title                     truncated-chat-name
+      :title                     (if group-chat
+                                   (utils/truncate-str chat-name 30)
+                                   @(re-frame/subscribe [:contacts/contact-name-by-identity chat-id]))
```

Comment:
![image](https://user-images.githubusercontent.com/11790366/83120163-f575c800-a0d0-11ea-8788-c053ddafc435.png)

Reply:
:D , this is tricky, I know it looks a bit funny, but I don't want to run the `subscribe` if it's not a one-to-one. If I pass a component it will be not applying the header style, so I am not sure what's another option.